datasets/inception_subset.py

```diff
@@ -0,0 +1,174 @@
+            return plant_inception_mod
+        else:
+            print('at least one of the mods is empty, check experiment mods!!!')
+            return None
+
+    def __plant_inception_all_mods(self, plant):
+        plant_inception_all_mods = {}
+        for mod in parameters.used_modalities:
+            plant_inception_all_mods[mod] = self.__plant_inception_one_mod(plant, mod)
+        return plant_inception_all_mods
+
+    def __getitem__(self, idx):
+        plant = idx % self.num_plants
+        sample = dict(
+            inception=self.__plant_inception_all_mods(plant),
+            label=labels[self.experiment][plant],
+            plant=plant)
+
+        return sample
+
+
+class InceptionSubset(Dataset):
+    def __init__(self, used_plants, experiment, root_dir, used_modalities, plants: List[int]):
+        self.labels = labels
+        self.experiment = experiment
+        self.root_dir = root_dir
+        self.modalities = used_modalities
+        self.data = used_plants
+        self.plants = plants
+        self.num_plants = len(plants)
+
+        print('self.plants  = ', self.plants)
+
+    def __len__(self):   # needs to be corrected to the number of items per plant for each modality
+        return self.num_plants
+
+    def __getitem__(self, idx):
+        plant = self.plants[idx % self.num_plants]
+        cycle = idx // self.num_plants
+        data = self.data[self.data.num_plants * cycle + plant]
+        data['plant'] = idx % self.num_plants
+        return data
+
+    @staticmethod
+    def random_split(modalities: Modalities):
+        indices = np.arange(modalities.num_plants)
+        plant_labels = np.asarray(labels[modalities.experiment])
+        #  https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.train_test_split.html
+        train_indices, test_indices = train_test_split(indices, train_size=parameters.train_ratio,
+                                                       stratify=plant_labels)
+        return InceptionSubset(modalities, train_indices, parameters.experiment, parameters.experiment_path,
+                               parameters.used_modalities), InceptionSubset(modalities, test_indices,
+                                                                            parameters.experiment,
+                                                                            parameters.experiment_path,
+                                                                            parameters.used_modalities)
+
+    @classmethod  # A generator
+    def cross_validation(cls: Type['InceptionSubset'], used_plants: Modalities) -> \
+            Iterator[Tuple['InceptionSubset', 'InceptionSubset']]:
+        # https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.StratifiedKFold.html
+        n_plants = parameters.n_plants
+        n_classes = parameters.n_classes
+        n_splits = parameters.n_splits
+        X = tuple(i for i in range(n_plants))  # Plants by order
```

Comment:
- [ ] no need for comprehension here
```suggestion
        X = tuple(range(n_plants))  # Plants by order
```